technic/lag.lua

```diff
@@ -0,0 +1,40 @@
+
+local samples = {}
+
+local index = 1
+
+minetest.register_globalstep(function(dtime)
+	-- Looping index; we don't need to know the order of the samples,
+	-- so we can just overwrite previous ones to save a bit of performance
+	samples[index] = dtime
+	if index < 100 then
+		index = index + 1
+	else
+		index = 1
+	end
+end)
+
+
+function technic.get_max_lag()
+	if #samples < 1 then
+		return 0
+	end
+	local max_lag = 0
+	for _,t in pairs(samples) do
+		if t > max_lag then
+			max_lag = t
+		end
+	end
+	return max_lag
+end
+
+function technic.get_avg_lag()
+	if #samples < 1 then
+		return 0
+	end
+	local avg_lag = 0
+	for _,t in pairs(samples) do
```

Comment:
Same here as previous to squeeze every bit of performance out of loop you can use ipairs over pairs because samples is quaranteed to be always indexed without holes.